src/utils.ts

```diff
@@ -69,6 +78,73 @@ const estimateInternalCallGas = async ({
   return applyGasCorrectionFactor(estimation, estimatedGasCorrectionFactor);
 };
 
+const estimatePaymentGas = async ({
+  internalEstimationCorrection,
+  estimatedGasCorrectionFactor,
+  preDeploySWAddress,
+  relayRequest,
+}: PaymentGasEstimationParams): Promise<BigNumber> => {
+  const {
+    request: { tokenContract, tokenAmount, to },
+    relayData: { callForwarder, gasPrice, feesReceiver },
+  } = relayRequest;
+
+  if (tokenAmount.toString() === '0') {
+    return constants.Zero;
+  }
+
+  let tokenOrigin: PromiseOrValue<string> | undefined;
+
+  if (isDeployRequest(relayRequest)) {
```

Comment:
Do we really need this check? It looks like we do the same before calling this function.

Reply:
Yes, since the two functions are different and can be called from different places.

Reply:
Sure, but it looks like in any part where we call it, we do the same before calling the function... I think we should change it in the future